include/kframework/infinite-gas.md

```diff
@@ -10,129 +10,65 @@ In particular, this means that `#gas(_) <Int #gas(_) => false`, and `#gas(_) <=I
-      requires Gsload       < SCHED > <=Int G
-       andBool Gsstoreset   < SCHED > <=Int G
-       andBool Gsstorereset < SCHED > <=Int G
-      [simplification]
-
-    rule Csstore(SCHED, _, _, _) <=Int G => true
-      requires Gsload       < SCHED > <=Int G
-       andBool Gsstoreset   < SCHED > <=Int G
-       andBool Gsstorereset < SCHED > <=Int G
-      [simplification]
-
-    rule 0 <=Int Cmem(_, N)              => true requires 0 <=Int N       [simplification]
-    rule         Cmem(_, N) <Int #gas(G) => true requires N  <Int #gas(G) [simplification]
-
-    rule 0 <=Int Cgascap(_, _, _, _)                           => true                                                                                    [simplification]
-    rule         Cgascap(_, GCAP, GAVAIL, GEXTRA) <Int #gas(G) => true requires GCAP <Int #gas(G) andBool GAVAIL <Int #gas(G) andBool GEXTRA <Int #gas(G) [simplification]
-
-    rule 0 <=Int Cextra(_, _, _, _)              => true [simplification]
-    rule         Cextra(_, _, _, _) <Int #gas(_) => true [simplification]
-    rule         Cextra(_, _, _, _) <Int pow256  => true [simplification]
-
-    rule 0 <=Int #allBut64th(_)                => true                          [simplification]
-    rule         #allBut64th(G)  <Int #gas(G') => true requires G <Int #gas(G') [simplification]
-
-    rule 0 <=Int _:ScheduleConst < _:Schedule >               => true [simplification]
-    rule         _:ScheduleConst < _:Schedule >  <Int #gas(_) => true [simplification]
-
-    rule 0 <=Int G up/Int _                => true requires 0 <=Int G        [simplification]
-    rule G         up/Int _  <Int #gas(G') => true requires G  <Int #gas(G') [simplification]
-
-    rule 0 <=Int Csload(_, _)       => true  [simplification]
-    rule Csload(_, _) <Int #gas(_)  => true  [simplification]
-    rule #gas(_) <Int Csload(_, _)  => false [simplification]
-    rule Csload(_, _) <=Int #gas(_) => true  [simplification]
-
+    // // Arithemtic operators apply to encapsulated Int 
+    // rule #gas(I1) + I2:Int  => #gas(I1 +Int I2) 
+    // rule I1:Int  + #gas(I2) => #gas(I1 +Int I2) 
+    // rule #gas(I1) - I2:Int  => #gas(I1 -Int I2) 
+    // rule I1:Int  - #gas(I2) => #gas(I1 -Int I2) 
+    // rule #gas(I1) * I2:Int  => #gas(I1 *Int I2) 
+    // rule I1:Int  * #gas(I2) => #gas(I1 *Int I2) 
+    // // rule #gas(I1) / I2:Int  => #gas(I1 /Int I2)  requires I2 =/=Int 0
+    // // rule I1:Int  / #gas(I2) => #gas(I1 /Int I2)  requires I2 =/=Int 0
+
+    // Comparison operators apply to infinity
+    rule  #gas(_)  < _:Int  => false 
+    rule  _:Int   < #gas(_) => true  
+    rule  #gas(_) <= _:Int  => false 
+    rule  _:Int  <= #gas(_) => true  
+    rule  #gas(_)  > _:Int  => true  
+    rule  _:Int   > #gas(_) => false 
+    rule  #gas(_) >= _:Int  => true  
+    rule  _:Int  >= #gas(_) => false 
+
+    rule <k>  G:Int ~> #deductGas => . ... </k> <gas> #gas(GAVAIL) => #gas(GAVAIL -Int G) </gas>
+
```

Comment:
```suggestion

    rule <k>  G:Int ~> #deductGas => . ... </k> <gas> #gas(GAVAIL) => #gas(GAVAIL -Int G) </gas>

```

We don't want to deplicate this rule, better ot just modify the semantics to use `G -Gas G'` in the semantics directly.